.github/workflows/compliance.yml

```diff
@@ -0,0 +1,55 @@
+# Licensed under the Apache-2.0 license
+
+name: RISC-V Compliance tests
+
+on:
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  compliance:
+    runs-on: ubuntu-24.04
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install dependencies
+      run: |
+        sudo apt update && \
+        sudo apt upgrade -y && \
+        sudo apt install -y \
+          device-tree-compiler \
+          gcc-multilib \
+          gcc-riscv64-unknown-elf \
+          opam \
+          z3
+    - name: Install RISC-V Python tools
+      run: |
+        pip install git+https://github.com/riscv/riscv-isac.git@dev && \
+        pip install git+https://github.com/riscv/riscof.git
+    - name: Install riscv-isa-sim
+      run: |
+        git clone https://github.com/riscv-software-src/riscv-isa-sim.git && \
+        cd riscv-isa-sim && \
+        mkdir build && \
+        cd build && \
```

Comment:
not a blocker: but could use pushd and popd instead of cd, so that we don't have to care about the relative paths when going back.

Reply:
Good call. I just copied this from a previous repo, but I can make that modification.